api/src/convert/archive_block.rs

```diff
@@ -1,115 +1,102 @@
-
-        let mut archive_block_v1 = blockchain::ArchiveBlockV1::new();
-        archive_block_v1.set_block(bc_block);
-        archive_block_v1.set_block_contents(bc_block_contents);
+impl From<&BlockData> for blockchain::ArchiveBlock {
+    fn from(src: &BlockData) -> Self {
+        let mut archive_block = blockchain::ArchiveBlock::new();
+        let archive_block_v1 = archive_block.mut_v1();
+        archive_block_v1.set_block(src.block().into());
+        archive_block_v1.set_block_contents(src.contents().into());
 
         if let Some(signature) = src.signature() {
-            let bc_signature = blockchain::BlockSignature::from(signature);
-            archive_block_v1.set_signature(bc_signature);
+            archive_block_v1.set_signature(signature.into());
+        }
+        if let Some(metadata) = src.metadata() {
+            archive_block_v1.set_metadata(metadata.into());
         }
-
-        let mut archive_block = blockchain::ArchiveBlock::new();
-        archive_block.set_v1(archive_block_v1);
 
         archive_block
     }
 }
 
 /// Convert from blockchain::ArchiveBlock --> mc_transaction_core::BlockData
-impl TryFrom<&blockchain::ArchiveBlock> for mc_transaction_core::BlockData {
+impl TryFrom<&blockchain::ArchiveBlock> for BlockData {
     type Error = ConversionError;
 
     fn try_from(src: &blockchain::ArchiveBlock) -> Result<Self, Self::Error> {
         if !src.has_v1() {
             return Err(ConversionError::ObjectMissing);
         }
+        let archive_block_v1 = src.get_v1();
 
-        let block = mc_transaction_core::Block::try_from(src.get_v1().get_block())?;
-
-        let block_contents =
-            mc_transaction_core::BlockContents::try_from(src.get_v1().get_block_contents())?;
+        let block = archive_block_v1.get_block().try_into()?;
+        let block_contents = BlockContents::try_from(archive_block_v1.get_block_contents())?;
 
-        let signature = src
-            .get_v1()
+        let signature = archive_block_v1
             .signature
             .as_ref()
-            .map(mc_transaction_core::BlockSignature::try_from)
+            .map(BlockSignature::try_from)
             .transpose()?;
-
         if let Some(signature) = signature.as_ref() {
             signature
                 .verify(&block)
                 .map_err(|_| ConversionError::InvalidSignature)?;
         }
 
-        if block.contents_hash != block_contents.hash() {
-            return Err(ConversionError::InvalidContents);
-        }
+        let metadata = archive_block_v1
+            .metadata
+            .as_ref()
+            .map(SignedBlockMetadata::try_from)
+            .transpose()?;
 
-        Ok(mc_transaction_core::BlockData::new(
-            block,
-            block_contents,
-            signature,
-        ))
+        if block.contents_hash == block_contents.hash() {
+            Ok(BlockData::new(block, block_contents, signature, metadata))
```

Comment:
If it's not already in one of the try statements above, the block id should be computed & matched. 

```suggestion
        if block.contents_hash == block_contents.hash() && block.is_block_id_valid() {
            Ok(BlockData::new(block, block_contents, signature, metadata))
```